### ISSUE TITLE
Dense3D

### ISSUE THREAD

**Issue by kmhallen** [l0g1x/DUO-Camera-ROS#11]
It appears the dense3d node mentioned in the README is empty. When do you expect this feature to be available?

**Comment by l0g1x** [comment]
@kmhallen  Yes, i havent been able to get around to it. DUO has sent me a beta version of the linux shared library for dense3D a couple days ago, so i should have a working version in the next 2-3weeks hopefully.

**Comment by gitunit** [comment]
do you also plan to use the duo calibration files? so far i find these superior to the ROS calibration.

**Comment by l0g1x** [comment]
Yes i am going to use the duo calibration files. The yaml format is different between the two anyway. Will hopefully have something done by end of today after school 

Krystian Gebis
University of Illinois at Chicago

**Comment by gitunit** [comment]
nice! if i can help, just tell me, we could speed up things. Though im new to ROS... ;)

**Comment by l0g1x** [comment]
Actually, if you would like to help i created a issue for something that i do need! #12 Thanks :) it isnt urgent, as it is mainly a "enhancement"

**Comment by gitunit** [comment]
ok, i will look into it.

**Comment by l0g1x** [comment]
DUO gave a me system.img for the Nvidia Jetson TK1 with the duo arm libraries loaded for beta testing purposes. having a little trouble flashing the image to SD card though. been working on this everyday since i got the image, hopefully will be able to get it to work soon.

**Comment by sak797** [comment]
Did you use the duo calibration files? Is there any way to convert them?

**Comment by l0g1x** [comment]
I finally free'd up some time to finish the rest of the driver, so in the next couple of days the driver will be able to use the duo calibration files, as well as utilize the dense3d portion

**Comment by l0g1x** [comment]
I have just found out from DUO that in the near upcoming release, the camera calibration will actually be placed on the camera device itself. That being said, there will no longer be a need for ros camera calibration files, nor duo dashboard calibration files. They are also planning to release much better calibration and api support for windows and linux as well.

**Comment by gitunit** [comment]
good news. however i can imagine this will take some time. :(

**Comment by l0g1x** [comment]
Release is scheduled within the next few days so no worries!

**Comment by gitunit** [comment]
do you mean the release from DUO3D? (that's at least what i meant) ;)

**Comment by l0g1x** [comment]
Yes from DUO :)

Krystian Gebis
University of Illinois at Chicago

**Comment by sak797** [comment]
great :) . Also I wanted to know if the dense3d library was utilized in the present version? I didn't see any code related to that

**Comment by sak797** [comment]
The package has been released, but I am facing issues while loading modules i.e while using **insmod duo.ko** and getting error **Invalid module format**. My kernel version is 3.16.0-41-generic and one in the documentation is 3.16.0-30 generic. Is that the problem?

**Comment by duo3d** [comment]
You might want to try and use '-f' option when calling insmod and see if that lets you load duo.ko module.

Alex

**Comment by l0g1x** [comment]
@duo3d I have the same issue except on 3.13.0-51-generic . I read in the readme.txt that at the top the kernel version was `Ubuntu 14.04.02 x64 (Kernel 3.16.0-30-generic)` so i tried updating my kernel like specified at this link: http://askubuntu.com/questions/506956/how-can-i-install-3-16-kernel-on-ubuntu-14-04 
but it updated me to `3.16.0-41-generic #55~14.04.1-Ubuntu` like @kanirudh54  , and still same issue as before.  I tried insmod -f, all as root, but still getting same issue.

**Comment by sak797** [comment]
I also had updated my kernel as it was older that 3.16 using the command *_sudo apt-get install linux-generic-lts-utopic xserver-xorg-lts-utopic libegl1-mesa-drivers-lts-utopic xserver-xorg-video-all-lts-utopic xserver-xorg-input-all-lts-utopic *_ from the [Ubuntu](https://wiki.ubuntu.com/TrustyTahr/ReleaseNotes) website.

**Comment by sak797** [comment]
When I tried **dmesg -x**, I got **kern  :warn  : [85848.000653] duo: disagrees about version of symbol module_layout**, hope that may be useful

**Comment by sak797** [comment]
I think the problem is due to the difference in **vermagic** number

**Comment by l0g1x** [comment]
@kanirudh54 i compiled from source the kernel module for `3.16.0-41-generic #55~14.04.1-Ubuntu` and it works now. You can download the duo.ko file here https://www.dropbox.com/s/0ujx4nm384pmtph/duo.ko?dl=0 . Just replace it with the one that you currently have in the folder.

**Comment by l0g1x** [comment]
@kanirudh54 are you able to run DUODashboard after you use the module i linked you?

**Comment by sak797** [comment]
Thanks, I haven't tried till now. I will try and let you know

**Comment by sak797** [comment]
@l0g1x , I am getting segmentation fault. This is the error message **kern  :info  : [  450.661925] DUODashboard[3201]: segfault at 8 ip 00007fd5df132ae0 sp 00007ffd08dfdf00 error 4 in libQt5Widgets.so.5.2.1[7fd5defc5000+5fb000]** . Also the system becomes unstable after starting the DUOCalibration .

**Comment by l0g1x** [comment]
@kanirudh54 im getting seg fault as well. I didnt get your error however (it didnt actually output any error, except "segmentation fault"), i just got a core dump which i sent to duo for them to look at.

**Comment by l0g1x** [comment]
I think it might have something to do with the qt libraries though, because ive seen something similar when i had both qt4 and qt5 installed. And it makes me believe it is something related to qt because i can run duo's firmware upgrade executable just fine (i dont believe it uses qt, or at least certain components that the dashboard and calibration exec's use). Looking into this now.

**Comment by sak797** [comment]
@l0g1x , even I got just segmentation fault. I have checked the message buffer of kernel which gave the extra information

**Comment by sak797** [comment]
@l0g1x @duo3d any updates?

**Comment by l0g1x** [comment]
Im still waiting on a fix from them as well. Last week when i talked with them it seemed as that they found the problem, but have been waiting for a new build since then. If i found out anything new ill post here

**Comment by sak797** [comment]
Thanks

**Comment by duo3d** [comment]
Hi Krystian,

I have a fix, I'm going to test it on the machine with no AVX support and
will send you the updated Linux package sometime today.

Regards,
Alex

**Comment by sak797** [comment]
@l0g1x @duo3d : any updates?
